src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -87,18 +93,25 @@ export async function setFirstDataSourceAsDefault(
   }
 }
 
-export function handleNoAvailableDataSourceError(
-  changeState: () => void,
-  notifications: ToastsStart,
-  application?: ApplicationStart,
-  callback?: (ds: DataSourceOption[]) => void
-) {
+export interface HandleNoAvailableDataSourceErrorProps {
+  changeState: () => void;
+  notifications: ToastsStart;
+  application?: ApplicationStart;
+  callback?: (ds: DataSourceOption[]) => void;
+  incompatibleDataSourcesExist: boolean;
```

Comment:
nit: might be easier to read to have required fields above optional

Reply:
Refactored